lib/react.dart

```diff
@@ -103,58 +104,204 @@ abstract class Component {
     _jsRedraw();
   }
 
-  /**
-   * set _nextState to newState
-   * and call React original setState method with no parameter
-   */
+  /// Set [_nextState] to provided [newState] value and force a re-render.
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-api.html#replacestate>
   void replaceState(Map newState) {
     Map nextState = newState == null ? {} : new Map.from(newState);
     _nextState = nextState;
+
     _jsRedraw();
   }
 
+  /// ReactJS lifecycle method that is invoked once, both on the client and server, immediately before the initial
+  /// rendering occurs.
+  ///
+  /// If you call [setState] within this method, [render] will see the updated state and will be executed only once
+  /// despite the [state] value change.
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#mounting-componentwillmount>
   void componentWillMount() {}
 
+  /// ReactJS lifecycle method that is invoked once, only on the client _(not on the server)_, immediately after the
+  /// initial rendering occurs.
+  ///
+  /// At this point in the lifecycle, you can access any [ref]s to the children of [rootNode].
+  ///
+  /// The [componentDidMount] method of child [Component]s is invoked _before_ that of parent [Component].
```

Comment:
`[Component]` probably doesn't have to be linked here—we're already inside `Component`.

https://www.dartlang.org/effective-dart/documentation/#avoid-redundantly-mentioning-types-in-doc-comments